ui/src/notebooks/index.ts

```diff
@@ -1,10 +1,17 @@
-import {FunctionComponent, ComponentClass} from 'react'
+import {FunctionComponent, ComponentClass, ReactNode} from 'react'
+
+export interface PipeContextProps {
+  children?: ReactNode
+}
+
+export type PipeData = any
 
 export interface PipeProp {
-  index: number
-  remove: () => void
-  moveUp: () => void
-  moveDown: () => void
+  data: PipeData
```

Comment:
i disconnected the relationship between pipes and notebook context here, so beware